src/frontend/komponenter/Fagsak/DataValidator.tsx

```diff
@@ -0,0 +1,36 @@
+const harFyltInnFødselsnummer = (filled: String): boolean => {
+    return /^\d{11}$/.test(filled.replace(' ', ''));
+};
+
+const fødselsnummerPassererMod10ogMod11Sjekk = (fn: String): boolean => {
+    const vektSifreMod10 = [3, 7, 6, 1, 8, 9, 4, 5, 2, 1];
+    const vektSifreMod11 = [5, 4, 3, 2, 7, 6, 5, 4, 3, 2, 1];
+
+    let sumMod10 = 0;
+    for (let i = 0; i < 10; i++) {
+        sumMod10 += Number.parseInt(fn.charAt(i), 10) * vektSifreMod10[i];
+    }
+
+    let sumMod11 = 0;
+    for (let i = 0; i < 11; i++) {
+        sumMod11 += Number.parseInt(fn.charAt(i), 10) * vektSifreMod11[i];
+    }
+
+    return sumMod10 % 11 === 0 && sumMod11 % 11 === 0;
+};
+
+enum ErrorId {
```

Comment:
I think we should introduce the IFelt from soknad-kontantstotte. But maybe not do everything in the reducer(since we dont have one).

So I think that all input fields in our app should be of type IFelt and contain the value, validation functions (which we import from a validators.ts file perhaps) and an optional error message.

In this way we can handle all input fields with one function which runs the validation and optionally sets an error message. Come talk to me if you are having trouble understanding the concept! 👍

Reply:
Sounds interesting! I will check the IFelt related code and refactor the implementation🛠